### ISSUE TITLE
Pitfall for `docker` newbies while create new image based on `opp_env` (all `opp_env` commands lead to errors

### ISSUE THREAD

**Issue by WolframStrauss** [omnetpp/opp_env#18]
Hi,
thank you so much for your project. It saves a lot of time.
I ran into a lot of errors while creating a docker image for our [CI](https://en.wikipedia.org/wiki/Continuous_integration). I tried the Omnet setup in an interactive container and it worked fine, e.g `which opp_env`.
Later the corresponding instruction in the `Dockerfile`, i.e. `RUN which opp_env`  led to an error. The reason is the limited scope of the docker [RUN](https://docs.docker.com/engine/reference/builder/#run) instruction, which means that e.g. path settings are gone as soon as the next `RUN` instruction is reached.
A solution is to prefix each `opp_env` entry with [source .profile](https://www.baeldung.com/linux/path-variable#persisting) (https://github.com/omnetpp/opp_env/blob/main/docker/Dockerfile).

As I mentioned I am a docker newbie, hence there might be errors in my explanations. Wolfram

**Comment by rhornig** [comment]
yes. opp_env is basically giving you a `nix develop` shell prompt loaded with a lot of state (lot of environment variables defined). You cannot avoid them so sourcing the .profile is necessary. Of course it's also possible to create a wrapper script around opp_env that does this but creating this just for the sake of Docker build files would be an overkill.

You can also execute all required opp_env commands in a single shell line (in a single RUN command) but that would create only a single layer in the docker file which would or would not be what you want.

Finally, when `opp_env` matured enough we may opt to automatically build Docker images from omnet using `opp_env`. Nix can also generate docker files directly so in future we might even add some command options to opp_env to produce a docker image directly (but this is not on our radar right now).

For now, just source `.profile` or do everything in a single RUN command.

Just out of curiosity, can you share your Docker file? I'm interested in what image are you using as base and how many and what opp_env commands you need for your CI image. I may add some instructions for future users who want to use it for docker.

**Comment by rhornig** [comment]
Closing this as there is nothing to do.

**Comment by WolframStrauss** [comment]
Here is my Dockerfile:
```dockerfile
# Based on: https://github.com/omnetpp/opp_env
ARG DOCKER_REGISTRY="ghcr.io"
# https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry
FROM ${DOCKER_REGISTRY}/omnetpp/opp_env:latest

# Prevent interactions during installation
ARG DEBIAN_FRONTEND=noninteractive

# `root` section: Force apt to retry downloads in case of connection failure.
USER root
RUN echo 'Acquire::Retries "10";' > /etc/apt/apt.conf.d/80-retries
# System update / upgrade
RUN apt-get -y update \
    && apt-get -y upgrade \
    && apt-get -y clean

# Have to be after `FROM ...`
ENV OMNETPP_VERSION="6.0.2"
ENV INET_VERSION="4.5.2"

# `opp_env` user section
USER ${USER}
# Should be: `Ubuntu 22.04 LTS`, /bin/bash, /home/opp_env
RUN more /etc/issue.net; echo $0; pwd

# Note: Each `RUN` line need `source .profile` to provide the necessary 
# paths, ... to use `opp_env` (https://www.baeldung.com/linux/path-variable)
RUN source .profile && which opp_env

# Omnet installation
RUN source .profile && mkdir workspace && cd workspace \
&& opp_env init && opp_env install omnetpp-${OMNETPP_VERSION}

# Run Oment test simulation. $USER from https://github.com/omnetpp/opp_env/blob/main/docker/Dockerfile
RUN source .profile && cd /home/${USER}/workspace/omnetpp-${OMNETPP_VERSION}/samples/tictoc/ \
&& opp_env run -c './tictoc -c Tictoc10 -u Cmdenv'
```

**Comment by rhornig** [comment]
Thanks. Pretty much what I was expecting. Once opp_env is considered stable, we will start generating similar docker images automatically up-to the last line. 

I don't see the point of running the last line during docker build. That line would be the actual program to be tested so it would be run during instatiation, not during building the image.

**Comment by WolframStrauss** [comment]
You are absolutely right.
